src/controllers/DoctorLoadedPatientController.java

```diff
@@ -86,37 +87,78 @@ private Command DeletePatientPrescription() {
-            } else {
-                doctorView.showDeletePrescriptionOutOfRangeError();
+            if (CheckUserInteger(deleteIndex, prescriptionDataList.size())) {
+                return;
             }
+            prescriptionManager.removePrescription(prescriptionDataList.get(deleteIndex));
+            doctorView.showSuccessfullyDeletedPrescription();
         };
     }
 
-/* PENDING IMPLEMENTATION IN PHASE 2
+
     private Command ViewPatientAppointments() {
         return (x) -> doctorView.viewAppointments(new AppointmentManager(getDatabase())
                 .getPatientAppointments(patientData));
     }
 
-    private Command getReport() {
-        return (x) -> new ReportManager(getDatabase()).getReportData(patientData);
+    private Command ViewPatientReports() {
+        return (x) -> {
+            ArrayList<ReportData> reportData = reportManager.getReportData(patientData);
+            if (reportData == null || reportData.size() == 0) {
+                doctorView.showNoReportsError();
+                return;
+            }
+            doctorView.viewAllReports(reportData);
+            Integer targetReport = doctorView.viewReportPrompt();
+            if (CheckUserInteger(targetReport, reportData.size())) {
+                return;
+            }
+            doctorView.viewReport(reportData.get(targetReport));
+        };
     }
 
-    private Command createReport() {
-        return (x) -> new ReportManager(getDatabase()).addReport(patientData, doctorData,
-                doctorView.reportDetailsPrompt().header(), doctorView.reportDetailsPrompt().body());
+    private Command CreatePatientReport() {
+        return (x) -> {
+            ReportDetails reportDetails = doctorView.reportDetailsPrompt();
+            reportManager.addReport(patientData, doctorData, reportDetails.header(), reportDetails.body());
+            doctorView.showReportCreationSuccess();
+        };
     }
 
-    private Command deleteReport() {
+    private Command DeletePatientReport() {
         return (x) -> {
+            ArrayList<ReportData> reportData = reportManager.getReportData(patientData);
+            if (reportData == null || reportData.size() == 0) {
+                doctorView.showNoReportsError();
+                return;
+            }
+
             Integer deleteIndex = doctorView.deleteReportPrompt(new ContactManager(getDatabase())
-                    .getContactData(patientData), new ReportManager(getDatabase()).getReportData(patientData));
-            getDatabase().getReportDatabase().remove(deleteIndex);
+                    .getContactData(patientData), reportData);
+
+            if (CheckUserInteger(deleteIndex, reportData.size())) {
+                return;
+            }
+            reportManager.deleteReport(reportData.get(deleteIndex));
+            doctorView.showReportDeletionSuccess();
+
         };
-    } */
+    }
 
+    private boolean CheckUserInteger(Integer deleteIndex, Integer size) {
```

Comment:
Please rename this method to something more descriptive like "isInvalidIndex" or something along those lines.